Script/install.sh

```diff
@@ -11,115 +10,30 @@ REPO_PATH=$(readlink -f $(dirname $0) | grep -E -o "^(.*?\\GrovePi)")
-	echo "2) Clone, build wiringPi in GrovePi/Script and install it"
-	echo "3) Removes I2C and SPI from modprobe blacklist /etc/modprobe.d/raspi-blacklist.conf"
-	echo "4) Adds I2C-dev, i2c-bcm2708 and spi-dev to /etc/modules"
-	echo "5) Installs gertboard avrdude_5.10-4_armhf.deb package"
-	echo "6) Runs gertboard setup"
-	echo "   - configures avrdude"
-	echo "   - downloads gertboard known boards and programmers"
-	echo "   - replaces avrsetup with gertboards version"
-	echo "   - in /etc/inittab comments out lines containing AMA0"
-	echo "   - in /boot/cmdline.txt removes: console=ttyAMA0,115200 kgdboc=ttyAMA0,115200 console=tty1"
-	echo "   - in /usr/share/arduino/hardware/arduino creates backup of boards.txt"
-	echo "   - in /usr/share/arduino/hardware/arduino creates backup of programmers.txt"
-	echo " "
+  echo " "
 	echo "Special thanks to Joe Sanford at Tufts University. This script was derived from his work. Thank you Joe!"
-	echo " "
-	echo "Raspberry Pi wil reboot after completion."
-	echo " "
-	echo " "
+  echo " "
 }
 
-check_root_user() {
-    if [[ $EUID -ne 0 ]]; then
-        feedback "FAIL!  This script must be run as such: sudo ./install.sh"
-        exit 1
-    fi
-    echo " "
-}
+install_dependencies() {
 
-check_internet() {
-    if ! quiet_mode ; then
-        feedback "Check for internet connectivity..."
-        feedback "=================================="
-        wget -q --tries=2 --timeout=20 --output-document=/dev/null https://raspberrypi.org
-        if [ $? -eq 0 ];then
-            echo "Connected to the Internet"
-        else
-            echo "Unable to Connect, try again !!!"
-            exit 0
-        fi
-    fi
-}
+    # the sudo apt-get update is already
+    # done by the script_tools installer in
+    # update_grovepi.sh
 
-install_dependencies() {
-    if ! quiet_mode ; then
-        sudo apt-get update
-    fi
     echo " "
-	feedback "Installing Dependencies"
-	echo "======================="
-	sudo apt-get install python-pip git libi2c-dev python-serial i2c-tools python-smbus python3-smbus arduino minicom python-dev -y
-	sudo apt-get purge python-rpi.gpio -y
-	sudo apt-get purge python3-rpi.gpio -y
-	sudo apt-get install python-rpi.gpio -y
-	sudo apt-get install python3-rpi.gpio -y
-       sudo apt-get install python-scipy -y
-       sudo apt-get install python3-scipy -y
-	sudo pip install -U RPi.GPIO
-  sudo pip2 install numpy
-  sudo pip3 install numpy
+  	feedback "Installing Dependencies"
+  	echo "======================="
+  	sudo apt-get install git libi2c-dev i2c-tools arduino minicom -y
+    sudo apt-get purge python-rpi.gpio -y
+  	sudo apt-get purge python3-rpi.gpio -y
+    sudo apt-get install python-pip python-smbus python-dev python-serial python-rpi.gpio python-scipy -y
+    sudo apt-get install python3-smbus python3-dev python3-rpi.gpio python3-scipy -y
+  	sudo pip install -U RPi.GPIO
+    sudo pip3 install -U RPi.GPIO
+    sudo pip2 install numpy
```

Comment:
Does numpy really have to be installed for GrovePi to work? Can't we just install it for the projects that actually need it?

Reply:
Yes. It's needed. We already have it installed for 8 months or so.

Reply:
Unless .. unless we re-write a part of the code to that's processed with numpy. It should be doable.